ffmpeg/lpms_ffmpeg.c

```diff
@@ -687,23 +897,32 @@ int process_in(struct input_ctx *ictx, AVFrame *frame, AVPacket *pkt)
   return ret;
 
 dec_flush:
+
+  // Flush and close decoder for non-CUDA
+
   if (ictx->vc) {
-    avcodec_send_packet(ictx->vc, NULL);
+    send_first(ictx);
+
+    // XXX sanity check this works as expected with SW encoding!
     ret = avcodec_receive_frame(ictx->vc, frame);
-    pkt->stream_index = ictx->vi; // XXX ugly?
-    if (!ret) return ret;
+    pkt->stream_index = ictx->vi;
+    if (!ret) {
+      if (is_flush_frame(frame)) ictx->flushed = 1;
+      return ret;
+    }
   }
   if (ictx->ac) {
```

Comment:
Can you describe logic here? It looks like we're getting here only if there is no video stream, or if there was error in `avcodec_receive_frame`. But why?

Reply:
That's basically correct. Note that the "error" case also indicates normal cases where there's no video frames available from the decoder - this is also signaled by the return code, eg EAGAIN or EOF.

This is just the way we flush the decoder. If there's a nonzero response type, we know there's no video frame to return. So we continue on to the audio decoder, and attempt to return an audio frame from there.

Reply:
Thanks! Maybe add comment explaining this to the code?

Reply:
Comment added in https://github.com/livepeer/lpms/pull/141/commits/f7fb9b6ea5cbcd08e760a40c0e03470f4376ebd6